castai/resource_workload_scaling_policy.go

```diff
@@ -0,0 +1,342 @@
+		return nil
+	}
+
+	delResp, err := client.WorkloadOptimizationAPIDeleteWorkloadScalingPolicyWithResponse(ctx, clusterID, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err := sdk.StatusOk(delResp); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceWorkloadScalingPolicyDiff(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	// Since tf doesn't support cross field validation, doing it here.
+	cpu := toResourcePolicies(d.Get("cpu").([]interface{})[0].(map[string]interface{}))
+	memory := toResourcePolicies(d.Get("memory").([]interface{})[0].(map[string]interface{}))
+
+	if err := validateArgs(cpu, "cpu"); err != nil {
+		return err
+	}
+	return validateArgs(memory, "memory")
+}
+
+func validateArgs(r sdk.WorkloadoptimizationV1ResourcePolicies, res string) error {
+	if r.Function == "QUANTILE" && len(r.Args) == 0 {
+		return fmt.Errorf("field %q: QUANTILE function requires args to be provided", res)
+	}
+	if r.Function == "MAX" && len(r.Args) > 0 {
+		return fmt.Errorf("field %q: MAX function doesn't accept any args", res)
+	}
+	return nil
+}
+
+func workloadScalingPolicyImporter(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	ids := strings.Split(d.Id(), "/")
+	if len(ids) != 2 || ids[0] == "" || ids[1] == "" {
+		return nil, fmt.Errorf("expected import id with format: <cluster_id>/<scaling_policy name or id>, got: %q", d.Id())
+	}
+
+	clusterID, id := ids[0], ids[1]
+	if err := d.Set(FieldClusterID, clusterID); err != nil {
+		return nil, fmt.Errorf("setting cluster id: %w", err)
+	}
+	d.SetId(id)
+
+	// Return if scaling policy ID provided.
+	if _, err := uuid.Parse(id); err == nil {
+		return []*schema.ResourceData{d}, nil
+	}
+
+	// Find scaling policy ID by name.
+	client := meta.(*ProviderConfig).api
+	resp, err := client.WorkloadOptimizationAPIListWorkloadScalingPoliciesWithResponse(ctx, clusterID)
+	if err := sdk.CheckOKResponse(resp, err); err != nil {
+		return nil, err
+	}
+
+	for _, sp := range resp.JSON200.Items {
+		if sp.Name == id {
+			d.SetId(sp.Id)
+			return []*schema.ResourceData{d}, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to find workload scaling policy with the following name: %v", id)
+}
+
+func toResourcePolicies(obj map[string]interface{}) sdk.WorkloadoptimizationV1ResourcePolicies {
+	out := sdk.WorkloadoptimizationV1ResourcePolicies{}
+
+	if v, ok := obj["function"].(string); ok {
+		out.Function = sdk.WorkloadoptimizationV1ResourcePoliciesFunction(v)
+	}
+	if v, ok := obj["args"].([]interface{}); ok && len(v) > 0 {
+		out.Args = toStringList(v)
+	}
+	if v, ok := obj["overhead"].(float64); ok {
+		out.Overhead = v
+	}
+	if v, ok := obj["apply_threshold"].(float64); ok {
+		out.ApplyThreshold = v
+	}
+
+	return out
+}
+
+func toResourceMap(p sdk.WorkloadoptimizationV1ResourcePolicies) []map[string]interface{} {
```

Comment:
Similar to other comments I would love to see a name with a more narrow meaning/context.